lib/quicly.c

```diff
@@ -2918,13 +2991,16 @@ static int handle_ack_frame(quicly_conn_t *conn, size_t epoch, quicly_ack_frame_
         0 /* this relies on the fact that we do not (yet) retransmit ACKs and therefore latest_rtt becoming UINT32_MAX */);
     /* OnPacketAckedCC */
     uint32_t cc_type = 0;
-    /* TODO (jri): this function should be called for every packet newly acked. */
-    if (quicly_loss_on_packet_acked(&conn->egress.loss, frame->largest_acknowledged)) {
-        cc_type = CC_RTO;
-        conn->egress.cc.in_first_rto = 0;
-    } else if (conn->egress.cc.in_first_rto) {
-        cc_type = CC_RTO_ERR;
-        conn->egress.cc.in_first_rto = 0;
+    /* TODO (jri): this function should be called for every packet newly acked. (kazuho) I do not think so;
+     * quicly_loss_on_packet_acked is NOT OnPacketAcked */
+    if (smallest_newly_acked != UINT64_MAX) {
```

Comment:
We may want to change this behavior as well... I think we're probably moving away from looking for newly acked packets in the ACK frame... This is also a change in PR 2040 and 1974.